### PR TITLE
Coverig the rendering part with unit tests.

### PR DESCRIPTION
**Pull request by etiennesegonzac**
Fixes issue #16

### CODE REVIEW

**Comment by etiennesegonzac** [comment]
Now on to debug travis :)

File: .jshintrc [hunk 1]
```diff
@@ -23,6 +23,8 @@
   "laxcomma": true,
   "proto": true,
 
+  "esnext": true,
```
Comment (etiennesegonzac) [inline .jshintrc:26]: we were missing some files in the linter, and removing this was actually too strict apparently.

File: fast-list.js [hunk 2]
```diff
@@ -106,7 +105,7 @@ FastList.prototype = {
     var itemPerScreen = geo.viewportHeight / geo.itemHeight;
     // Taking into account the will-change budget multiplier from
     // layout/base/nsDisplayList.cpp#1193
-    geo.maxItemCount = Math.floor(itemPerScreen * 2.4);
+    geo.maxItemCount = Math.floor(itemPerScreen * 2.8);
```
Comment (etiennesegonzac) [inline fast-list.js:108]: this change snuck in as part of an earlier commit. but the issue I was trying to fix is actually gaia or gecko related but not from the list.

File: fast-list.js [hunk 3]
```diff
@@ -240,9 +239,7 @@ FastList.prototype = {
   },
 
   updateListHeight: function() {
```
Comment (etiennesegonzac) [inline fast-list.js:241]: always called as part of a mutation block already

File: fast-list.js [hunk 4]
```diff
@@ -268,29 +265,27 @@ FastList.prototype = {
     var source = this.source;
     var template = this._templateSection;
 
-    return schedule.mutation(function() {
```
Comment (etiennesegonzac) [inline fast-list.js:271]: same here

**Comment by etiennesegonzac** [comment]
`TOTAL: 42 SUCCESS`

Ready to review @wilsonpage  :)

File: test/fast-list-test.js [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+      assert.equal(container.style.overflowY, 'scroll');
+    });
+
+    test('it removes the tempates from the DOM', function() {
+      assert.isNull(container.querySelector('ul li'));
+      assert.isNull(container.querySelector('ul section'));
+    });
+
+    test('it attaches a direct block to the scroll event', function() {
+      sinon.assert.calledOnce(scheduler.attachDirect);
```
Comment (wilsonpage) [inline test/fast-list-test.js:107]: NIT: could add second assertion with more detail?

`sinon.assert.calledWith(scheduler.attachDirect, container, 'scroll');`

File: test/fast-list-test.js [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+      assertCurrentlyRenderedWindow(15, 35);
+    });
+
+    suite('really fast >', function() {
+      setup(function() {
+        container.scrollTop += 1200; // 2.5 viewports
+        scheduler.attachDirect.yield();
+      });
+
+      test('it skips rendering (same content)', function() {
```
Comment (wilsonpage) [inline test/fast-list-test.js]: What does 'skips rendering' mean?

File: test/fast-list-test.js [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+        assertCurrentlyRenderedWindow(15, 35);
+      });
+
+      suite('multiple times >', function() {
+        setup(function() {
+          container.scrollTop += 480; // 1 viewport
+          scheduler.attachDirect.yield();
```
Comment (wilsonpage) [inline test/fast-list-test.js:180]: What does the `.yield()` do?

**Comment by wilsonpage** [comment]
Looks fine, ship it!

File: test/fast-list-test.js (hunk 5 above)
Comment (etiennesegonzac) [inline test/fast-list-test.js:107]: oh yeah I definitely wanted to do that :)

File: test/fast-list-test.js (hunk 6 above)
Comment (etiennesegonzac) [inline test/fast-list-test.js]: tried to change the wording a bit, does it help?

File: test/fast-list-test.js (hunk 7 above)
Comment (etiennesegonzac) [inline test/fast-list-test.js:180]: triggers the callback that was given to `attachDirect`

**Pull request merged**